src/core/algorithms/nar/des/des.cpp

```diff
@@ -0,0 +1,87 @@
+
+#include "algorithms/nar/value_range.h"
+#include "config/names_and_descriptions.h"
+#include "config/option_using.h"
+#include "config/tabular_data/input_table/option.h"
+#include "model/types/types.h"
+
+namespace algos::des {
+using model::ValueRange;
+
+DES::DES() : NARAlgorithm({}) {
+    using namespace config::names;
+    RegisterOptions();
+}
+
+void DES::RegisterOptions() {
+    DESBORDANTE_OPTION_USING;
+
+    DifferentialStrategy default_strategy = DifferentialStrategy::rand1Bin;
+    RegisterOption(Option{&population_size_, kPopulationSize, kDPopulationSize, 100u});
+    RegisterOption(
+            Option{&num_evaluations_, kMaxFitnessEvaluations, kDMaxFitnessEvaluations, 1000u});
+    RegisterOption(Option{&differential_options_.differential_scale, kDifferentialScale,
+                          kDDifferentialScale, 0.5});
+    RegisterOption(Option{&differential_options_.crossover_probability, kCrossoverProbability,
+                          kDCrossoverProbability, 0.9});
+    RegisterOption(Option{&differential_options_.differential_strategy, kDifferentialStrategy,
+                          kDDifferentialStrategy, default_strategy});
+}
+
+void DES::MakeExecuteOptsAvailable() {
+    using namespace config::names;
+    MakeOptionsAvailable({kPopulationSize, kMaxFitnessEvaluations, kDifferentialScale,
+                          kCrossoverProbability, kDifferentialStrategy});
+}
+
+FeatureDomains DES::FindFeatureDomains(TypedRelation const* typed_relation) {
+    std::vector<std::shared_ptr<ValueRange>> feature_domains;
+    feature_domains.reserve(typed_relation->GetNumColumns());
+    for (size_t i = 0; i < typed_relation->GetNumColumns(); i++) {
+        std::shared_ptr<ValueRange> domain = CreateValueRange(typed_relation->GetColumnData(i));
+        feature_domains.push_back(std::move(domain));
+    }
+    return feature_domains;
+}
+
+std::vector<EncodedNAR> DES::GetRandomPopulationInDomains(FeatureDomains const& domains) const {
+    std::vector<EncodedNAR> population(population_size_,
+                                       EncodedNAR(domains, typed_relation_.get()));
+    auto compare_by_fitness = [](EncodedNAR const& a, EncodedNAR const& b) {
+        return a.GetQualities().fitness > b.GetQualities().fitness;
+    };
+    std::sort(population.begin(), population.end(), compare_by_fitness);
+    return population;
+}
+
+EncodedNAR DES::MutatedIndividual(std::vector<EncodedNAR> const& population, size_t at) {
+    MutationFunction diff_func =
+            EnumToMutationStrategy(differential_options_.differential_strategy);
+    return (*diff_func)(population, at, differential_options_);
+}
+
+unsigned long long DES::ExecuteInternal() {
+    FeatureDomains feature_domains = FindFeatureDomains(typed_relation_.get());
+    std::vector<EncodedNAR> population = GetRandomPopulationInDomains(feature_domains);
+
+    for (unsigned i = 0; i < num_evaluations_; i++) {
+        size_t candidate_i = i % population_size_;
```

Comment:
You mix `unsinged`, `size_t` and `int` in one statement. Generally, it's not safe (see, for example, https://github.com/Nekrolm/ubbook/blob/master/numeric/integer_promotion.md). Is '`int` the best type for `population_size_`? I guess, **size** cannot be negative.
The same is for `num_evaluations_` in this class.